src/protocol/codecs/topic.rs

```diff
@@ -0,0 +1,112 @@
+use std::io::{self, ErrorKind};
+
+use bytes::BytesMut;
+use tokio_util::codec::Decoder;
+
+use crate::protocol::invalid_data;
+
+/// Topic is a key-value string pair.
+pub struct Topic {
+    /// Key.
+    pub key: String,
+
+    /// Value.
+    pub value: String,
+}
+
+#[derive(Default, Clone)]
+pub struct TopicCodec {}
```

Comment:
```suggestion
pub struct TopicCodec;
```

Reply:
Oh, that's possible. TIL ✅

Clippy could warn about it, though, right?

Reply:
actually no, I think both ways are equally acceptable; I just prefer the "leaner" way :smiley: